openedx_learning/apps/authoring/containers/api.py

```diff
@@ -0,0 +1,334 @@
+    """
+    return EntityListRow.objects.create(
+        entity_list=entity_list,
+        entity_id=entity_pk,
+        order_num=order_num,
+        draft_version_id=draft_version_pk,
+        published_version_id=published_version_pk,
+    )
+
+
+def create_entity_list_with_rows(
+    entity_pks: list[int],
+    draft_version_pks: list[int | None],
+    published_version_pks: list[int | None],
+) -> EntityList:
+    """
+    Create a new entity list with rows.
+
+    Args:
+        entity_pks: The IDs of the publishable entities that the entity list rows reference.
+        order_nums: The order_nums of the entity list rows in the entity list.
+        draft_version_pks: The IDs of the draft versions of the entities (PublishableEntityVersion) that the entity list rows reference.
+        published_version_pks: The IDs of the published versions of the entities (PublishableEntityVersion) that the entity list rows reference.
+
+    Returns:
+        The newly created entity list.
+    """
+    entity_list = create_entity_list()
+    order_nums = range(len(entity_pks))
+    for entity_pk, order_num, draft_version_pk, published_version_pk in zip(
+        entity_pks, order_nums, draft_version_pks, published_version_pks
+    ):
+        create_entity_list_row(
+            entity_list=entity_list,
+            entity_pk=entity_pk,
+            order_num=order_num,
+            draft_version_pk=draft_version_pk,
+            published_version_pk=published_version_pk,
+        )
+    return entity_list
+
+
+def create_container_version(
+    container_pk: int,
+    version_num: int,
+    title: str,
+    publishable_entities_pk: list[int],
+    entity: PublishableEntity,
+    created: datetime,
+    created_by: int | None,
+) -> ContainerEntityVersion:
+    """
+    Create a new container version.
+
+    Args:
+        container_pk: The ID of the container that the version belongs to.
+        version_num: The version number of the container.
+        title: The title of the container.
+        publishable_entities_pk: The IDs of the members of the container.
+        entity: The entity that the container belongs to.
+        created: The date and time the container version was created.
+        created_by: The ID of the user who created the container version.
+
+    Returns:
+        The newly created container version.
+    """
+    with atomic():
+        publishable_entity_version = publishing_api.create_publishable_entity_version(
+            entity.pk,
+            version_num=version_num,
+            title=title,
+            created=created,
+            created_by=created_by,
+        )
+        # This implementation assumes:
+        # 1. We are creating the first version of the container, so the defined list is the same as the initial list.
+        # 2. The frozen list is empty because this is the first version.
+        # 3. Published and draft versions are always the latest for all members.
```

Comment:
This is also a naive approach, this should also be compliant with the requirements I mentioned in the cover letter.